### PR TITLE
Add persist_stats_cf_ to DBImpl and relative API

### PR DESCRIPTION
**Pull request by GlitterIsMe**
Add persist_stats_cf_ to DBImpl and relative API for using persist_stats_cf to store GC stats

### CODE REVIEW

**Comment by Connor1996** [comment]
Please give a brief description of which part you have changed comparing to origin PR.

File: db/db_impl.cc [hunk 1]
```diff
@@ -2658,7 +2671,7 @@ Status DestroyDB(const std::string& dbname, const Options& options,
     InfoLogPrefix info_log_prefix(!soptions.db_log_dir.empty(), dbname);
     for (const auto& fname : filenames) {
       if (ParseFileName(fname, &number, info_log_prefix.prefix, &type) &&
-        type != kDBLockFile) {  // Lock file will be deleted at end
+          type != kDBLockFile) {  // Lock file will be deleted at end
```
Comment (Connor1996) [inline db/db_impl.cc]: Please keep the origin format

**Comment by Connor1996** [comment]
@GlitterIsMe Other places still have format issue 🤣

**Comment by GlitterIsMe** [comment]
> @GlitterIsMe Other places still have format issue 🤣
I used clang-format(google) to format the my code, but it modified other place I didn't changed...

**Comment by yiwu-arbug** [comment]
CI is failing, have a look?

**Comment by yiwu-arbug** [comment]
Please list what are the commits cherry-picked from upstream.

**Pull request closed without merging**